src/sentinel1_reader/sentinel1_burst_slc.py

```diff
@@ -322,17 +324,26 @@ class represents a polynomial function of range
            'x' and azimuth 'y'
         """
 
+        rdr_grid = self.as_isce3_radargrid()
+
         lines, samples = self.shape
         x = np.arange(0, samples, xstep, dtype=int)
         y = np.arange(0, lines, ystep, dtype=int)
-        xx, yy = np.meshgrid(x, y)
+        x_mesh, y_mesh = np.meshgrid(x, y)
 
         # Estimate azimuth carrier
         az_carrier = compute_az_carrier(self, self.orbit,
                                         offset=offset,
-                                        position=(yy, xx))
+                                        position=(y_mesh, x_mesh))
+
+        # Change index mesh to range/az mesh?
+        if not index_as_coord:
+            rg = self.starting_range + (x + 1) * self.range_pixel_spacing
+            az = rdr_grid.sensing_start + (y + 1) * self.azimuth_time_interval
+            x_mesh, y_mesh = np.meshgrid(rg, az)
+
         # Estimate azimuth carrier polynomials
-        az_carrier_poly = polyfit(xx.flatten() + 1, yy.flatten() + 1,
+        az_carrier_poly = polyfit(x_mesh.flatten(), y_mesh.flatten(),
                                   az_carrier.flatten(), az_order,
                                   rg_order)
```

Comment:
I think this part should slightly change to:

```
if index_as_coord:
            az_carrier_poly = polyfit(x_mesh.flatten()+1, y_mesh.flatten()+1,
                                  az_carrier.flatten(), az_order,
                                  rg_order)
else:
            rg = self.starting_range + (x + 1) * self.range_pixel_spacing
            az = rdr_grid.sensing_start + (y + 1) * self.azimuth_time_interval
            x_mesh, y_mesh = np.meshgrid(rg, az)

            # Estimate azimuth carrier polynomials
            az_carrier_poly = polyfit(x_mesh.flatten(), y_mesh.flatten(),
                                  az_carrier.flatten(), az_order,
                                  rg_order)
```

This way I could consistently eval the carrier when it is a function of index or azimuth range with the following test:

```
rdrGrid = bursts[0].as_isce3_radargrid()

azCarr1 = bursts[0].get_az_carrier_poly(index_as_coord=True)
azCarr2 = bursts[0].get_az_carrier_poly()

i_az = 10
i_rng = 5
print(azCarr1.eval(i_az,i_rng))

az = rdrGrid.sensing_start + (i_az)/rdrGrid.prf
rng = rdrGrid.starting_range + (i_rng)*rdrGrid.range_pixel_spacing

print(azCarr2.eval(az,rng))

```
which results in the following close enough values:

11019.073192749373
11019.073192726628

@vbrancat would you please try again your gslc test with this minor change to the reader in this PR.